xmtp_api_grpc/src/streams/try_from_item.rs

```diff
@@ -0,0 +1,40 @@
+//! Maps a `TryStream` with a different type that implements `TryFrom<T>` for the streams item
+
+use std::{marker::PhantomData, task::{ready, Poll}};
+use futures::{TryStream, Stream};
+use pin_project_lite::pin_project;
+
+pin_project! {
+    pub struct TryFromItem<S, T> {
+        #[pin] inner: S,
+        _marker: PhantomData<T>,
+    }
+}
+
+/// Wrap a `TryStream<T>` such that it converts its 'item' to T
+pub fn try_from_stream<S, T>(s: S) -> TryFromItem<S, T> {
```

Comment:
these adapter streams help translate the proto types to the concrete libxmtp type (xmtp_proto::types), should be moved to xmtp_api_d14n